From b18589f8697f1d9f9ce6f97ec75a7026c14d4584 Mon Sep 17 00:00:00 2001
From: Ronan <ronan.lemartret@iot.bzh>
Date: Fri, 6 Jan 2017 17:55:47 +0100
Subject: [PATCH] fix gcc 6.2 build

 * prevent error, when gcc 6.2 is used:
     "case label is not an integer constant expression"
   from file :
     "services/spd/opteed/opteed_main.c"
   code:
     "case TEESMC_OPTEED_RETURN_ON_DONE
      case TEESMC_OPTEED_RETURN_RESUME_DONE
      ..."
   source error:
    code:
      ./services/spd/opteed/teesmc_opteed_macros.h
    from TEESMC_OPTEED_RV: (SMC_TYPE_FAST << FUNCID_TYPE_SHIFT)

   "
   #define SMC_TYPE_FAST 1
   #define FUNCID_TYPE_SHIFT 31
   "

 * ref:
   http://c0x.coding-guidelines.com/6.5.7.html
   https://github.com/ARM-software/tf-issues/issues/438

Signed-off-by: Ronan <ronan.lemartret@iot.bzh>
---
 atf/include/bl31/runtime_svc.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/atf/include/bl31/runtime_svc.h
+++ b/atf/include/bl31/runtime_svc.h
@@ -57,7 +57,7 @@
 #define SMC_64				1
 #define SMC_32				0
 #define SMC_UNK				0xffffffff
-#define SMC_TYPE_FAST			1
+#define SMC_TYPE_FAST			1U
 #define SMC_TYPE_STD			0
 #define SMC_PREEMPTED		0xfffffffe
 /*******************************************************************************
